subiquity/server/controllers/ad.py

```diff
@@ -37,12 +38,45 @@
 FIELD = "Domain Controller name"
 
 
+class DcPingStrategy:
+    # Consider staging in the snap for future proof.
+    cmd = "/usr/sbin/realm"
+    arg = "discover"
+
+    async def ping(self, address: str) -> AdDomainNameValidation:
+        cp = await arun_command([self.cmd, self.arg, address], env={})
+        if cp.returncode:
+            return AdDomainNameValidation.REALM_NOT_FOUND
+
+        return AdDomainNameValidation.OK
+
+
+class StubDcPingStrategy(DcPingStrategy):
+    """ For testing purpose. This class doesn't talk to the network.
+        Instead its response follows the following rule:
+
+        - addresses starting with "r" return REALM_NOT_FOUND;
+        - addresses starting with any other letter return OK. """
+    async def ping(self, address: str) -> AdDomainNameValidation:
+        if address[0] == 'r':
+            return AdDomainNameValidation.REALM_NOT_FOUND
+
+        return AdDomainNameValidation.OK
```

Comment:
@jpnurmi this is what you've requested in the first meeting (based on the Ubuntu Pro mocking) isn't it?

Reply:
yes, this is perfect. :) thanks!